docs/root/configuration/listeners/network_filters/connection_limit_filter.rst

```diff
@@ -0,0 +1,106 @@
+-  The filter maintains an atomic counter of active connection count. It has a max connections limit value based on the configured total number of connections.
+   When a new connection request comes, the filter tries to increment the connection counter. The connection is allowed if the counter is less than the max connections limit, otherwise the connection gets rejected.
+   When an active connection is closed, the filter decrements the active connection counter.
+-  The filter does not stop connection creation but will close the connections that were accepted but were deemed as overlimit.
+-  **Slow rejection:** The filter can stop reading from the connection and close it after a delay instead of rejecting it right away or letting requests go through before the rejection.
+   This way we can prevent a malicious entity from opening new connections while draining their resources.
+
+.. note::
+  In the current implementation each filter chain has an independent connection limit.
+
+Algorithm
+---------
+
+The filter will use the reference counting algorithm to keep trace of active connection count.
+
+1. Active connection count < max connection limit:
+
+   -  Increment count for a new allowed connection.
+   -  Decrement count when a connection closes.
+
+2. Active connection count >= max connection limit:
+
+   -  Close the new connection request after configured delay time.
+
+API Reference
+-------------
+
+[extensions.filters.network.connection_limit.v3.ConnectionLimit proto]
+
+{
+
+**“stat_prefix”**: “…”,
+
+**“max_connections”**: “…”,
+
+**“delay”**: “{…}”,
+
+**“runtime_enabled”**: “{…}”
+
+}
+
+**stat_prefix**
+
+(`string <https://developers.google.com/protocol-buffers/docs/proto#scalar>`_)
+The prefix to use when emitting statistics.
+
+**max_connections**
+
+(`uint64 <https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.UInt64Value>`_)
+The limit supplied in max connections.
+
+**delay**
+
+(`Duration <https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#duration>`_)
+The delay in milliseconds for the slow rejection connections. If not set, this defaults to 0ms.
+
+**runtime_enabled**
+
+(`config.core.v3.RuntimeFeatureFlag <https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/base.proto#envoy-v3-api-msg-config-core-v3-runtimefeatureflag>`_)
+Runtime flag that controls whether the filter is enabled or not. If not specified, defaults to enabled.
+
+.. _config_network_filters_connection_limit_stats:
+
+Statistics
+----------
+
+The connection limit filter outputs statistics in the *<stat_prefix>.connection_limit* namespace.
+
++-----------------------+-----------------------+-----------------------+
+| **Name**              | **Type**              | **Description**       |
++=======================+=======================+=======================+
+| limited_connections   | Counter               | Total number of       |
```

Comment:
closed_connections?

Reply:
I prefer to use the limited_connections since the connections may be in the delay closing duration.